tests/lib/filterAtfs.test.ts

```diff
@@ -81,4 +82,62 @@ describe('filterAtfs library unit tests', () => {
       expect(result).toEqual(atfs);
     });
   });
+
+  describe('removeAtfsWithNoGeolocationData', () => {
+    const log: Logger = logger.create('apiRequestId', 'correlationId');
+    test('ATFs with no geolocation property are removed from the list', () => {
+      const atfs: AuthorisedTestingFacility[] = <AuthorisedTestingFacility[]><unknown>[
+        { availability: { isAvailable: false } },
+        { availability: { isAvailable: false } },
+        { geoLocation: { lat: 4, long: 2 } },
+        { geoLocation: { lat: 2, long: 4 } },
+      ];
+      const result: AuthorisedTestingFacility[] = filterAtfs.removeAtfsWithNoGeolocationData(atfs, log);
+      const expectedAtfs: AuthorisedTestingFacility[] = <AuthorisedTestingFacility[]><unknown>[
+        { geoLocation: { lat: 4, long: 2 } },
+        { geoLocation: { lat: 2, long: 4 } },
+      ];
+      expect(result).toEqual(expectedAtfs);
+    });
+
+    test('ATFs with geolocation and a latitude and longitude are not removed', () => {
+      const atfs: AuthorisedTestingFacility[] = <AuthorisedTestingFacility[]><unknown>[
+        { geoLocation: { lat: 4, long: 2 } },
+        { geoLocation: { lat: 2, long: 4 } },
+      ];
+      const result: AuthorisedTestingFacility[] = filterAtfs.removeAtfsWithNoGeolocationData(atfs, log);
+      expect(result).toEqual(atfs);
+    });
+
+    test('ATFs with missing latitude property are removed from the list', () => {
+      const atfs: AuthorisedTestingFacility[] = <AuthorisedTestingFacility[]><unknown>[
+        { geoLocation: { long: 5 } },
+        { geoLocation: { lat: null, long: 4 } },
+        { geoLocation: { lat: undefined, long: 3 } },
+      ];
+      const result: AuthorisedTestingFacility[] = filterAtfs.removeAtfsWithNoGeolocationData(atfs, log);
+      expect(result).toEqual([]);
+    });
+
+    test('ATFs with missing longitude property are removed from the list', () => {
```

Comment:
These all have long properties  - I think fixture is wrong

Reply:
Any value in having a mixed fixture 
- good and bad objects

Reply:
good spot! Added in a mixed fixture for this and also the missing latitude scenario.
Also added a spy onto the logger warning and assertions that its called for each removal